src/tree/images/ImageTreeItem.ts

```diff
@@ -72,13 +72,75 @@ export class ImageTreeItem extends AzExtTreeItem {
 
             // error code 409 is returned for conflicts like the image is used by a running container or another image.
             // Such errors are not really an error, it should be treated as warning.
-            if (parsedError.errorType === '409') {
-                ext.outputChannel.appendLog(localize('vscode-docker.tree.images.warning', 'Warning: {0}', parsedError.message));
-                // eslint-disable-next-line @typescript-eslint/no-floating-promises
-                window.showWarningMessage(parsedError.message);
+            if (parsedError.errorType === '409' || parsedError.errorType === 'conflict') {
+                await this.handleImageDeleteConflict(context);
+
+                // Throw a UserCancelledError instead since this isn't a real error
+                throw new UserCancelledError();
             } else {
                 throw error;
             }
         }
     }
+
+    private async handleImageDeleteConflict(context: IActionContext): Promise<void> {
+        const childTags = await this.getChildImageTags();
+
+        if (childTags.length > 0) {
+            const message = localize('vscode-docker.tree.images.dependentImages', 'Image {0} cannot be removed because the following image(s) depend on it and must be removed first: {1}', this.fullTag, childTags.join(', '));
+            const removeChildren = localize('vscode-docker.tree.images.removeDependentImages', 'Remove Dependent Images');
+
+            const allImageNodes = await ext.imagesRoot.getFlattenedCachedChildren(context);
+
+            // Since childTags comes as depth-first search, deleting in this specific order should work, as we'll be targeting leaf nodes first
+            const dependentImageNodes: ImageTreeItem[] = childTags.map(childFullTag => {
+                return allImageNodes.find(n => n instanceof ImageTreeItem && n.fullTag === childFullTag) as ImageTreeItem
+            });
+
+            // eslint-disable-next-line @typescript-eslint/no-floating-promises
+            window.showWarningMessage(message, ...[removeChildren]).then(response => {
+                if (response === removeChildren) {
+                    // eslint-disable-next-line @typescript-eslint/no-floating-promises
+                    commands.executeCommand('vscode-docker.images.remove', dependentImageNodes[0], dependentImageNodes);
```

Comment:
Can the delete be handled internally instead of invoking the command? This will inflate the telemetry count.

Reply:
Invoking the command was very much intentional, because it properly deals with the tree nodes.

Reply:
How about telemetry? Is there a way to distinguish user action Vs we internally trigger.

Reply:
Good thought. I'll see what we can do.